common/dbconfig.h

```diff
@@ -0,0 +1,60 @@
+// Copyright 2021-present Pinterest. All Rights Reserved.
+//
+// @author Premkumar (prem@pinterest.com)
+//
+
+#pragma once
+
+#include <memory>
+#include <string>
+#include <unordered_map>
+
+#include "common/config.h"
+#include "common/jsoncpp/include/json/json.h"
+
+/**
+ * {
+ *   "dataset" : {
+ *      // dataset level configs
+ *        "ad_query_join" : {
+ *              "ack_mode" : 1,
+ *       }
+ *    }
+ * }
+ */
+
+namespace common {
+
+struct DBConfig {
+  std::unordered_map<std::string, Config> dataSetConfigMap;
+};
+
+class DBConfigManager {
```

Comment:
Are all public interfaces thread-safe? Can you please add some comments for that?

Reply:
+1

Reply:
Not all public interfaces are thread safe. The initialization happens via a single instance (from ServiceConfig) and atomically swapped. The shared_ptr after atomic load, takes care of thread safety automatically

Reply:
updated with comments